spec/models/scanned_resource_spec.rb

```diff
@@ -174,4 +174,33 @@
       expect(subject.valid?).to eq false
     end
   end
+
+  describe "#check_completion" do
+    it "completes record when state changes to 'complete'" do
+      allow(subject).to receive("state_changed?").and_return true
+      subject.state = 'complete'
+      expect(subject).to receive(:complete_record)
+      expect { subject.check_completion }.to change { ActionMailer::Base.deliveries.count }.by(1)
+    end
+    it "does not complete record when state doesn't change" do
+      allow(subject).to receive("state_changed?").and_return false
+      subject.state = 'complete'
+      expect(subject).not_to receive(:complete_record)
+      expect { subject.check_completion }.not_to change { ActionMailer::Base.deliveries.count }
+    end
+    it "does not complete record when state isn't 'complete'" do
+      subject.state = 'pending'
+      expect(subject).not_to receive(:complete_record)
+      expect { subject.check_completion }.not_to change { ActionMailer::Base.deliveries.count }
+    end
+    it "does not overwrite existing identifier" do
+      allow(subject).to receive("state_changed?").and_return true
+      subject.state = 'complete'
+      subject.identifier = '1234'
+      expect(subject).to receive(:complete_record)
+      expect(subject).not_to receive("identifier=")
+      expect { subject.check_completion }.to change { ActionMailer::Base.deliveries.count }.by(1)
```

Comment:
Same as above: this test fails, but the functionality appears to be working for me when I run the app.